server/controllers/finance/cashflow.js

```diff
@@ -0,0 +1,476 @@
+/**
+ * Cashflow Controller
+ *
+ *
+ * This controller is responsible for processing cashflow report.
+ *
+ * @module finance/cashflow
+ *
+ * @requires node-uuid
+ * @requires lib/db
+ * @requires lib/util
+ * @requires lib/errors/NotFound
+ * @requires lib/errors/BadRequest
+ * @requires finance/journal/cash
+ */
+
+'use strict';
+
+const uuid = require('node-uuid');
+const db   = require('../../lib/db');
+const util = require('../../lib/util');
+const rm   = require('../../lib/ReportManager');
+const NotFound = require('../../lib/errors/NotFound');
+const BadRequest = require('../../lib/errors/BadRequest');
+
+// expose to the API
+exports.report = report;
+exports.document = document;
+
+/**
+ * @function report
+ * @desc This function is responsible of generating the cashflow data for the report
+ */
+function report (req, res, next) {
+  let params = req.query;
+
+  processingCashflowReport(params)
+  .then(result => {
+    res.status(200).json(result);
+  })
+	.catch(next);
+}
+
+/** processingCashflowReport */
+function processingCashflowReport(params) {
+  let glb = {};
+
+  // get all periods for the the current fiscal year
+	return getPeriods(params.dateFrom, params.dateTo)
+	.then(function (periods) {
+		// get the closing balance (previous fiscal year) for the selected cashbox
+    if (!periods.length) {
+      throw new BadRequest('Periods not found due to a bad date interval', 'ERRORS.BAD_DATE_INTERVAL');
+    }
+		glb.periods = periods;
+		return closingBalance(params.account_id, glb.periods[0].start_date);
+	})
+	.then(function (balance) {
+		if (!balance.length) { balance[0] = { balance: 0, account_id: params.account_id }; }
+		glb.openningBalance = balance[0];
+		return queryIncomeExpense(params);
+	})
+	.then(function (result) {
+		return groupingByMonth(glb.periods, result);
+	})
+	.then(groupingIncomeExpenseByPeriod)
+	.then(function (flows) {
+		return { openningBalance : glb.openningBalance, flows : flows };
+	});
+}
+
+/**
+ * @function queryIncomeExpense
+ * @param {object} params
+ * @param {object} dateFrom The stating date to considerate
+ * @param {object} dateTo The stop date to considerate
+ * @description returns incomes and expenses data in a promise
+ */
+function queryIncomeExpense (params, dateFrom, dateTo) {
+	if (params && dateFrom && dateTo) {
+		params.dateFrom = dateFrom;
+		params.dateTo = dateTo;
+	}
+	var requette =
+			'SELECT BUID(`t`.`uuid`) AS uuid, `t`.`trans_id`, `t`.`trans_date`, `a`.`number`, `a`.`label`, SUM(`t`.`debit_equiv`) AS debit_equiv,  ' +
```

Comment:
Could you use ES6 multiline strings instead of `'`?  It's much easier to read.  You can remove all the backticks (\`) from the query itself as well - they do not add anything of value.